provider/marathon/marathon_test.go

```diff
@@ -934,440 +700,565 @@ func TestMarathonGetProtocol(t *testing.T) {
+			provider := &Provider{}
+			actual := provider.getPassHostHeader(c.application)
+			if actual != c.expected {
+				t.Fatalf("actual %q, expected %q", actual, c.expected)
+			}
+		})
 	}
 }
 
-func TestMarathonGetEntryPoints(t *testing.T) {
-	provider := &Provider{}
+func TestMarathonMaxConnAmount(t *testing.T) {
+	cases := []struct {
+		desc        string
+		application marathon.Application
+		expected    int64
+	}{
+		{
+			desc:        "label missing",
+			application: application(),
+			expected:    math.MaxInt64,
+		},
+		{
+			desc: "non-integer value",
+			application: application(labels(map[string]string{
+				types.LabelBackendMaxconnAmount: "foobar",
+			})),
+			expected: math.MaxInt64,
+		},
+		{
+			desc: "label existing",
+			application: application(labels(map[string]string{
+				types.LabelBackendMaxconnAmount: "32",
+			})),
+			expected: 32,
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.desc, func(t *testing.T) {
+			t.Parallel()
+			provider := &Provider{}
+			actual := provider.getMaxConnAmount(c.application)
+			if actual != c.expected {
+				t.Fatalf("actual %d, expected %d", actual, c.expected)
+			}
+		})
+	}
+}
+
+func TestMarathonGetMaxConnExtractorFunc(t *testing.T) {
+	cases := []struct {
+		desc        string
+		application marathon.Application
+		expected    string
+	}{
+		{
+			desc:        "label missing",
+			application: application(),
+			expected:    "request.host",
+		},
+		{
+			desc: "label existing",
+			application: application(labels(map[string]string{
+				types.LabelBackendMaxconnExtractorfunc: "client.ip",
+			})),
+			expected: "client.ip",
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.desc, func(t *testing.T) {
+			t.Parallel()
+			provider := &Provider{}
+			actual := provider.getMaxConnExtractorFunc(c.application)
+			if actual != c.expected {
+				t.Fatalf("actual %q, expected %q", actual, c.expected)
+			}
+		})
+	}
+}
+
+func TestMarathonGetLoadBalancerMethod(t *testing.T) {
+	cases := []struct {
+		desc        string
+		application marathon.Application
+		expected    string
+	}{
+		{
+			desc:        "label missing",
+			application: application(),
+			expected:    "wrr",
+		},
+		{
+			desc: "label existing",
+			application: application(labels(map[string]string{
+				types.LabelBackendLoadbalancerMethod: "drr",
+			})),
+			expected: "drr",
+		},
+	}
+
+	for _, c := range cases {
+		c := c
+		t.Run(c.desc, func(t *testing.T) {
+			t.Parallel()
+			provider := &Provider{}
+			actual := provider.getLoadBalancerMethod(c.application)
+			if actual != c.expected {
+				t.Fatalf("actual %q, expected %q", actual, c.expected)
+			}
+		})
+	}
+}
 
-	applications := []struct {
+func TestMarathonGetCircuitBreakerExpression(t *testing.T) {
+	cases := []struct {
+		desc        string
+		application marathon.Application
+		expected    string
+	}{
+		{
+			desc:        "label missing",
+			application: application(),
+			expected:    "NetworkErrorRatio() > 1",
```

Comment:
This part is somewhat confusing: While we do return a default value, we [only include circuit-breaker settings](https://github.com/containous/traefik/blob/c53be185f4c71d41c1b71cd80a78d1613a970b65/templates/marathon.tmpl#L22) when [the label is actually set](https://github.com/containous/traefik/blob/c53be185f4c71d41c1b71cd80a78d1613a970b65/provider/marathon/marathon.go#L369-L372).

It might make sense to change the default return value to an empty string to improve clarity. Let's consider that for a future PR.